test/common/schemaform/fields/AutosuggestField.unit.spec.jsx

```diff
@@ -273,4 +299,124 @@ describe('<AutosuggestField>', () => {
+    const props = {
+      uiSchema: {
+        'ui:options': {
+          getOptions,
+          queryForResults: true
+        }
+      },
+      schema: { type: 'string' },
+      formContext: { reviewMode: false },
+      idSchema: { $id: 'id' },
+      onChange: () => {},
+      onBlur: () => {}
+    };
+    const wrapper = mount(<AutosuggestField {...props}/>);
+
+    // Search for 'ir'
+    const input = wrapper.find('input');
+    input.simulate('focus');
+    input.simulate('change', {
+      target: {
+        value: 'ir'
+      }
+    });
+
+    // Check that getOptions was called with the form data
+    const args = getOptions.secondCall.args;
+    expect(args[0]).to.eql('ir');
+    expect(args[1]).to.be.a('function');
+  });
+
+
+  it('should use the results of getOptions as the field\'s enum options', (done) => {
+    const props = {
+      uiSchema: {
+        'ui:options': {
+          getOptions: queryForOptions,
+          queryForResults: true
+        }
+      },
+      schema: { type: 'string' },
+      formContext: { reviewMode: false },
+      idSchema: { $id: 'id' },
+      onChange: () => {},
+      onBlur: () => {}
+    };
+    const wrapper = mount(<AutosuggestField {...props}/>);
+
+    setTimeout(() => {
+      expect(wrapper.state('options')).to.eql(options);
+    }, 100);
+
+    setTimeout(() => {
+      // Search for 'ir'
+      const input = wrapper.find('input');
+      input.simulate('focus');
+      input.simulate('change', {
+        target: {
+          value: 'ir'
+        }
+      });
+    }, 110);
+
+    setTimeout(() => {
+      expect(wrapper.state('options')).to.eql([
+        { id: 1, label: 'first' },
+        { id: 3, label: 'third' }
+      ]);
+      done();
+    }, 210);
+  });
+
+
+  // The stringifyFormReplacer will send the label to the api instead of the id if freeInput is
+  //  true in the formData
+  it('should set a freeInput property to true in the formData if freeInput is true in ui:options', () => {
+    const validationSpy = sinon.spy();
+    const props = {
+      uiSchema: {
+        'ui:options': {
+          freeInput: true,
+          labels: {
+            AL: 'Label 1',
+            BC: 'Label 2'
+          }
+        },
+        'ui:validations': [validationSpy]
+      },
+      schema: {
+        type: 'string',
+        'enum': ['AL', 'BC']
+      },
+      formContext: { reviewMode: false },
+      idSchema: { $id: 'id' },
+      onChange: () => {},
+      onBlur: () => {}
+    };
+    const wrapper = mount(<AutosuggestField {...props}/>);
+
+    // Input something not in options
+    const input = wrapper.find('input');
+    input.simulate('focus');
+    input.simulate('change', {
+      target: {
+        value: 'asdf'
+      }
+    });
+
+    setTimeout(() => {
+      const fieldData = validationSpy.firstCall.args[1];
+      expect(fieldData.freeInput).to.be.true;
+    });
+  });
+
+
+  it.skip('should fill in other formData when a suggestion is selected', () => {});
```

Comment:
Skipping this for now because it's arguably not necessary for MVP, but I want to keep it on our radar.